src/hooks/UrqlUseSubscription.re

```diff
@@ -17,20 +17,21 @@ type useSubscriptionArgs = {
   context: UrqlClient.ClientTypes.partialOperationContext,
 };
 
+type executeSubscriptionJs;
+
 [@bs.module "urql"]
 external useSubscriptionJs:
   (useSubscriptionArgs, option((option('acc), Js.Json.t) => 'acc)) =>
-  array(UrqlTypes.jsResponse('ret, 'extensions)) =
+  (UrqlTypes.jsResponse('ret, 'extensions), executeSubscriptionJs) =
```

Comment:
One thing to note here – this type can actually be expressed non-abstractly. It's the `executeSubscription` function, which is unfortunately not well documented in the `urql` docs. You have to look at the source for the version we're binding to (currently 1.5.1) to know: https://github.com/FormidableLabs/urql/blob/v1.5.1/src/hooks/useSubscription.ts#L55-L80.

The use cases for `executeSubscription` in the wild aren't many; I chatted with Phil Pluckthun about this today (also on `urql` core team) and it's mostly for feature parity with the other hooks. All this being said, I'll tackle this in a follow up PR; more just a good thing to know about, and thanks for finding this and bringing it up.

Reply:
👍